artview/plugins/io.py

```diff
@@ -0,0 +1,201 @@
+"""
+tree_view.py
+"""
+
+# Load the needed packages
+import code
+import pyart
+
+import sys
+import os
+
+path = os.path.dirname(sys.modules[__name__].__file__)
+path = os.path.join(path, '...')
+sys.path.insert(0, path)
+
+import artview
+
+from ..core import Component, Variable, common, QtGui, QtCore, componentsList
+
+open_functions = [
+    pyart.io.read,
+    pyart.io.read_grid,
+    pyart.io.read_grid_mdv,
+    pyart.aux_io.read_d3r_gcpex_nc,
+    pyart.aux_io.read_gamic,
+    pyart.aux_io.read_kazr,
+    pyart.aux_io.read_noxp_iphex_nc,
+    pyart.aux_io.read_odim_h5,
+    pyart.aux_io.read_pattern,
+    pyart.aux_io.read_radx]
+
+
+class AdvancedIO(Component):
+    '''
+    Open an interactive python console so the direct manipulation
+    '''
+
+    Vradar = None  #: see :ref:`shared_variable`
+    Vgrid = None  #: see :ref:`shared_variable`
+
+    @classmethod
+    def guiStart(self, parent=None):
+        '''Graphical interface for starting this class.'''
+        return self(), False
+
+    def __init__(self, dirIn=None, name="AdvancedIO", parent=None):
+        '''Initialize the class to create the interface.
+
+        Parameters
+        ----------
+        [Optional]
+        dirIn: string
+            Initial directory path to open.
+        name : string
+            Field Radiobutton window name.
+        parent : PyQt instance
+            Parent instance to associate to this class.
+            If None, then Qt owns, otherwise associated with parent PyQt
+            instance.
+        '''
+        super(AdvancedIO, self).__init__(name=name, parent=parent)
+        self.dirView = DirView(dirIn)
+        self.setCentralWidget(self.dirView)
+        self.dirView.opened.connect(self.open)
+        self.dirView.customContextMenuRequested.connect(self.contextMenu)
+        self.Vradar = Variable(None)
+        self.Vgrid = Variable(None)
+        self.sharedVariables = {"Vradar": None,
+                                "Vgrid": None}
+        self.connectAllVariables()
+        self.show()
+
+    def open(self, path):
+        ''' Slot of :py:attr:`DirView.opened` '''
+        # try several open
+        print ("open: %s" % path)
```

Comment:
This print statement might duplicate the RadarDisplay or GridDisplay message of file opening.